src/rust/src/orchard_keys_ffi.rs

```diff
@@ -95,6 +145,26 @@ pub extern "C" fn orchard_incoming_viewing_key_to_address(
     Box::into_raw(Box::new(key.address_at(diversifier_index)))
 }
 
+#[no_mangle]
+pub extern "C" fn orchard_incoming_viewing_key_decrypt_diversifier(
+    key: *const IncomingViewingKey,
+    addr: *const Address,
+    j_ret: *mut [u8; 11],
+) -> bool {
+    let key =
+        unsafe { key.as_ref() }.expect("Orchard incoming viewing key pointer may not be null.");
+    let addr = unsafe { addr.as_ref() }.expect("Orchard raw address pointer may not be null.");
+    let j_ret = unsafe { &mut *j_ret };
```

Comment:
If `j_ret` is a null pointer then this is undefined behaviour. (In practice, it will cause a SIGSEGV on Posix platforms, but that isn't guaranteed by Rust.)

Reply:
```suggestion
    assert!(!j_ret.is_null());
    let j_ret = unsafe { &mut *j_ret };
```

Reply:
Gah, I'd tried to catch all these usages in @nuttycom's patchset, but I missed one. I'll go though and double-check the diff. Here I want to use `unsafe { j_ret.as_mut() }.expect("must not be null")` like elsewhere.